docs/domain-names.md

```diff
@@ -0,0 +1,66 @@
+# Domain Names
+
+In order to support different scenarios that customers have, it is possible to specify a custom domain and hostnames for the machines that are built as a result of these templates.
+
+There are three patterns that are supported:
+
+1. No custom domain name and is not a Managed App
+
+In this case the servers will have a fully qualified domain name (FQDN) based on the prefix, server type, unique string and the domain for the region in Azure that was set. For example
+
+```
+Prefix: rjs
+Server Type: chef
+Unique String: fg6t
+Azure Region: eastus
+
+FQDN: rjs-chef-fg6t.eastus.cloudapp.azure.com
+```
+
+In this case the servers will utilise their own self signed certificates.
+
+2. Managed App and no custom domain
+
+Here the Managed App domain is `managedautomate.io` and the machines will be named accordingly.
+
+```
+Prefix: rjs
+Server Type: chef
+Unique String: fg6t
+
+FQDN: rjs-chef-fg6t.managedautomate.io
+```
+
+During configuration the servers will attempt to get an SSL certificate from Lets Encrypt using the FQDN of the machine.
+
+A cronjob is configured to run everyday to automatically renew the certificates as they are only valid for 90 days. As `certbot` is using the standalone mode it needs to shutdown the server proceses when a renewal has been successful.
+
+| Server | Command |
+|---|---|
+| Automate | `certbot renew --pre-hook "chef-automate stop" --post-hook "chef-automate start"` |
+| Chef | `certbot renew --pre-hook "chef-server-ctl stop nginx" --post-hook "chef-server-ctl start nginx`
+
+By default the cron is set to run every day at 0030, however this can be changed using the parameters and the Portal UI.
```

Comment:
How will stopping Automate & Chef Server every 24 hours affect customers who have runs or backups set to perform at these times?

As an aside, with the 90 day TTL of the certificates, is renewing every 24hr necessary? Couldn't this be performed once per week, or even once per month?

Reply:
The `certbot renew` command will check to see if the certificate needs renewing, and only if it does will it stop and restart the services. This is built into the function.

It is run every 24 hours so that it catches the day that the certificate may expire. It will renew certs that have less that 30 days to run.